failed to solve: useing MySQL 8.4.0 (2024-04-30, LTS Release)

### Description:
<!-- A clear and concise description of what the bug is. -->

When using mysql latest version，mysql server wolud not working. If change version to 8.3, would be work successfully.

```
# clean all data
docker system prune -a -f --volumes
sudo rm -rf ~/.docker
sudo rm -rf ~/.laradock

# downgrade to 8.3
- MYSQL_VERSION=latest
+ MYSQL_VERSION=8.3
```

### Expected Behavior:
<!-- A clear and concise description of what you expected to happen. -->

> - Important Change: The deprecated mysql_native_password authentication plugin is now disabled by default. It can be enabled by starting MySQL with the new --mysql-native-password=ON server option, or by adding mysql_native_password=ON to the [mysqld] section of your MySQL configuration file.

[Changes in MySQL 8.4.0 (2024-04-30, LTS Release)](https://dev.mysql.com/doc/relnotes/mysql/8.4/en/news-8-4-0.html)

### Context information:

**Output of `git rev-parse HEAD`**

```
0d213effcc06de905c6c24e48a76eab5a2451554
```

**Output of `docker version`**

```
Client: Docker Engine - Community
 Version:           26.1.1
 API version:       1.45
 Go version:        go1.21.9
 Git commit:        4cf5afa
 Built:             Tue Apr 30 11:47:53 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          26.1.1
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.9
  Git commit:       ac2de55
  Built:            Tue Apr 30 11:47:53 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.31
  GitCommit:        e377cd56a71523140ca6ae87e30244719194a521
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker-compose version`**

```
Docker Compose version v2.27.0
```

**System info: Mac, Windows or Linux. Include which disto/version**

```
Linux Mint 21.3 (Virginia)
```

### Steps to reproduce the issue:
<!--- How can we reproduce the issue? -->

1. run `docker compose up --build --detach nginx mysql phpmyadmin redis workspace`
2. check `docker compose exec mysql bash` . It will got `service "mysql" is not running` message
3. see logs  `docker logs laradock-mysql-1`
```
2024-05-02 02:22:05+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.4.0-1.el8 started.
2024-05-02 02:22:05+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-05-02 02:22:05+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.4.0-1.el8 started.
2024-05-02 02:22:05+00:00 [Note] [Entrypoint]: Initializing database files
2024-05-02T02:22:05.564082Z 0 [System] [MY-015017] [Server] MySQL Server Initialization - start.
2024-05-02T02:22:05.564971Z 0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2024-05-02T02:22:05.565006Z 0 [System] [MY-013169] [Server] /usr/sbin/mysqld (mysqld 8.4.0) initializing of server in progress as process 81
2024-05-02T02:22:05.565849Z 0 [Warning] [MY-013242] [Server] --character-set-server: 'utf8' is currently an alias for the character set UTF8MB3, but will be an alias for UTF8MB4 in a future release. Please consider using UTF8MB4 in order to be unambiguous.
2024-05-02T02:22:05.572406Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-05-02T02:22:05.886166Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2024-05-02T02:22:08.248848Z 0 [ERROR] [MY-000067] [Server] unknown variable 'default-authentication-plugin=mysql_native_password'.
2024-05-02T02:22:08.249158Z 0 [ERROR] [MY-013236] [Server] The designated data directory /var/lib/mysql/ is unusable. You can remove all files that the server added to it.
2024-05-02T02:22:08.249166Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-05-02T02:22:09.042714Z 0 [System] [MY-015018] [Server] MySQL Server Initialization - end.
```
### Stacktrace & Additional info:

```
{paste here}
```

--- a/mysql/Dockerfile
+++ b/mysql/Dockerfile
@@ -15,4 +15,10 @@ COPY my.cnf /etc/mysql/conf.d/my.cnf
 
 RUN chmod 0444 /etc/mysql/conf.d/my.cnf
 
+RUN if [ ${MYSQL_VERSION} > '8.4.0-0.000' ]; then \
+  echo 'mysql_native_password=on' >> /etc/mysql/conf.d/my.cnf \
+else \
+  echo 'default-authentication-plugin=mysql_native_password' >> /etc/mysql/conf.d/my.cnf \
+;fi
+
 EXPOSE 3306

--- a/mysql/my.cnf
+++ b/mysql/my.cnf
@@ -8,5 +8,4 @@
 [mysqld]
 sql-mode="STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION"
 character-set-server=utf8
-default-authentication-plugin=mysql_native_password
 innodb_use_native_aio=0

